### PR TITLE
2820 collect citizen data

### PR DESCRIPTION
**Pull request by omerzam**
# Description

Collect Citizen data for dashboard

About # ([2820](https://app.zenhub.com/workspaces/gooddollar-5d50833888a83c6880d3e345/issues/gooddollar/gooddapp/2820))

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes.

# Checklist:
- [ ] PR title matches follow: (Feature|Bug|Chore) Task Name
- [ ] My code follows the style guidelines of this project
- [ ] I have followed all the instructions described in the initial task (check Definitions of Done)
- [ ] I have performed a self-review of my own code
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] I have added reference to a related issue in the repository
- [ ] I have added a detailed description of the changes proposed in the pull request. I am as descriptive as possible, assisting reviewers as much as possible.
- [ ] I have added screenshots related to my pull request (for frontend tasks)
- [ ] I have pasted a gif showing the feature.
- [ ] @mentions of the person or team responsible for reviewing proposed changes

### CODE REVIEW

File: fuse/src/mapping.ts [hunk 1]
```diff
@@ -114,12 +117,95 @@ export function handleUBICalculated(event: UBICalculated): void {
...
+export function handleWithdrawFromDao(event: WithdrawFromDao): void { }
+
+export function handleWhitelistedAdded(event: WhitelistedAdded): void {
+  log.info('handleWhitelistedAdded event.params.account {}', [event.params.account.toHexString()])
+  let citizen = Citizen.load(event.params.account.toHex())
+
+  if (citizen == null) {
+    citizen = new Citizen(event.params.account.toHex())
+    citizen.claimStreak = BigInt.fromI32(1)
+    citizen.longestClaimStreak = BigInt.fromI32(1)
```
Comment (sirpy) [inline fuse/src/mapping.ts]: why? if it all it should be 0

File: fuse/src/mapping.ts [hunk 2]
```diff
@@ -114,12 +117,95 @@ export function handleUBICalculated(event: UBICalculated): void {
...
+export function handleWhitelistedRemoved(event: WhitelistedRemoved): void {
+  log.info('handleWhitelistedRemoved event.params.account {}', [event.params.account.toHex()])
+  let citizen = Citizen.load(event.params.account.toHex())
+
+  if (citizen == null) {
+    citizen = new Citizen(event.params.account.toHex())
+  }
+
+  // TODO: should claim streak be reset to zero when removing citizen?
+  citizen.claimStreak = BigInt.fromI32(0)
```
Comment (sirpy) [inline fuse/src/mapping.ts]: no, redundant

File: fuse/src/one-time-payment-mapping.ts [hunk 3]
```diff
@@ -16,10 +16,16 @@ export function handlePaymentWithdraw(event: PaymentWithdraw): void {
...
+    citizenTo.inTransactions = citizenTo.inTransactions.plus(BigInt.fromI32(1))
     citizenTo.inTransactionsClean = citizenTo.inTransactionsClean.plus(BigInt.fromI32(1))
 
+    citizenTo.inTransactionsValue = citizenTo.inTransactionsValue.plus(event.params.amount)
     citizenTo.inTransactionsValueClean = citizenTo.inTransactionsValueClean.plus(event.params.amount)
+
+    citizenTo.totalTransactions = citizenTo.totalTransactions.plus(BigInt.fromI32(1))
     citizenTo.totalTransactionsClean = citizenTo.totalTransactionsClean.plus(BigInt.fromI32(1))
+
+    citizenTo.totalTransactionsValue = citizenTo.totalTransactionsValue.plus(event.params.amount)
```
Comment (sirpy) [inline fuse/src/one-time-payment-mapping.ts]: dont update non clean values here

File: fuse/src/one-time-payment-mapping.ts [hunk 4]
```diff
@@ -29,9 +35,16 @@ export function handlePaymentWithdraw(event: PaymentWithdraw): void {
...
+    citizenFrom.outTransactions = citizenFrom.outTransactions.plus(BigInt.fromI32(1))
     citizenFrom.outTransactionsClean = citizenFrom.outTransactionsClean.plus(BigInt.fromI32(1))
+
+    citizenFrom.outTransactionsValue = citizenFrom.outTransactionsValue.plus(event.params.amount)
     citizenFrom.outTransactionsValueClean = citizenFrom.outTransactionsValueClean.plus(event.params.amount)
+
+    citizenFrom.totalTransactions = citizenFrom.totalTransactions.plus(BigInt.fromI32(1))
     citizenFrom.totalTransactionsClean = citizenFrom.totalTransactionsClean.plus(BigInt.fromI32(1))
+
+    citizenFrom.totalTransactionsValue = citizenFrom.totalTransactionsValue.plus(event.params.amount)
```
Comment (sirpy) [inline fuse/src/one-time-payment-mapping.ts]: dont update non clean values here

**Review by sirpy — CHANGES_REQUESTED** [review]

File: fuse/src/mapping.ts (hunk 2 above)
Comment (omerzam) [reply fuse/src/mapping.ts]: done

File: fuse/src/one-time-payment-mapping.ts (hunk 3 above)
Comment (omerzam) [reply fuse/src/one-time-payment-mapping.ts]: what do you mean by non-clean values?

File: fuse/src/mapping.ts (hunk 1 above)
Comment (omerzam) [reply fuse/src/mapping.ts]: done

**Pull request merged**